chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -351,33 +351,33 @@
 
     variables(
       prompt_from_template: proc { |user, cwd|
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", user) \
-          .gsub("CWD", cwd) \
-          .gsub("SUFFIX", "${prompt_suffix}") \
-          .gsub("ALIAS", aliaz) \
-          .gsub("HOST", node["hostname"]) \
-          .gsub("FQDN", node["fqdn"])
+        node["provisioner"]["shell_prompt"].to_s. \
+          gsub("USER", user). \
+          gsub("CWD", cwd). \
+          gsub("SUFFIX", "${prompt_suffix}"). \
+          gsub("ALIAS", aliaz). \
+          gsub("HOST", node["hostname"]). \
+          gsub("FQDN", node["fqdn"])
       },
 
       zsh_prompt_from_template: proc {
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", "%{\\e[0;31m%}%n%{\\e[0m%}") \
-          .gsub("CWD", "%{\\e[0;35m%}%~%{\\e[0m%}") \
-          .gsub("SUFFIX", "%#") \
-          .gsub("ALIAS", "%{\\e[0;35m%}#{aliaz}%{\\e[0m%}") \
-          .gsub("HOST", "%{\\e[0;35m%}#{node["hostname"]}%{\\e[0m%}") \
-          .gsub("FQDN", "%{\\e[0;35m%}#{node["fqdn"]}%{\\e[0m%}")
+        node["provisioner"]["shell_prompt"].to_s. \
+          gsub("USER", "%{\\e[0;35m%}%n%{\\e[0m%}"). \
+          gsub("CWD", "%{\\e[0;31m%}%~%{\\e[0m%}"). \
+          gsub("SUFFIX", "%#"). \
+          gsub("ALIAS", "%{\\e[0;31m%}#{aliaz}%{\\e[0m%}"). \
+          gsub("HOST", "%{\\e[0;31m%}#{node["hostname"]}%{\\e[0m%}"). \
+          gsub("FQDN", "%{\\e[0;31m%}#{node["fqdn"]}%{\\e[0m%}")
       },
 
       bash_prompt_from_template: proc {
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", "\\[\\e[01;31m\\]\\u\\[\\e[0m\\]") \
-          .gsub("CWD", "\\[\\e[01;31m\\]\\w\\[\\e[0m\\]") \
-          .gsub("SUFFIX", "${prompt_suffix}") \
-          .gsub("ALIAS", "\\[\\e[01;35m\\]#{aliaz}\\[\\e[0m\\]") \
-          .gsub("HOST", "\\[\\e[01;35m\\]#{node["hostname"]}\\[\\e[0m\\]") \
-          .gsub("FQDN", "\\[\\e[01;35m\\]#{node["fqdn"]}\\[\\e[0m\\]")
+        node["provisioner"]["shell_prompt"].to_s. \
+          gsub("USER", "\\[\\e[01;35m\\]\\u\\[\\e[0m\\]"). \
+          gsub("CWD", "\\[\\e[01;35m\\]\\w\\[\\e[0m\\]"). \
```

Comment:
I wonder if there was a reason why cwd has the same color as user in bash, but a different one in zsh. Added in cc726563 by @tboerger